pandas/core/reshape/reshape.py

```diff
@@ -75,10 +75,15 @@ def __init__(self, values, index, level=-1, value_columns=None,
                  fill_value=None):
 
         self.is_categorical = None
+        self.is_sparse = is_sparse(values)
         if values.ndim == 1:
             if isinstance(values, Categorical):
                 self.is_categorical = values
                 values = np.array(values)
+            elif self.is_sparse:
+                # XXX: Makes SparseArray *dense*, but it's supposedly
```

Comment:
A comment with a mild TODO hint at whomever will be refactoring the whole thing eventually to take the note into consideration.